### ISSUE TITLE
Documentation crosslinking

### ISSUE THREAD

**Issue by xstos** [Azure-Samples/active-directory-dotnet-admin-restricted-scopes-v2#27]
Hi, i've found documentation like [this](https://docs.microsoft.com/en-us/azure/app-service/samples-powershell?toc=%2fazure%2fapp-service%2fenvironment%2ftoc.json)

But then we have all these code samples like [this](https://github.com/Azure-Samples/active-directory-dotnet-admin-restricted-scopes-v2/blob/master/AppCreationScripts/Configure.ps1)

Is there a central documentation index for all this stuff somewhere to help people trying to learn? 

My end goal is to create an idempotent azure AD appregistration that can create an app from scratch, export the secrets and whatever else, and setup the tens of application permissions on behalf of our customers (and ourselves). Ideally it will also be able to delete an existing app, recreate it, and update whatever config to our backend automatically using some kind of REST endpoint. I'm surprised that there isn't a protocol for doing this between backends and azure itself, as all this manual moving of configuration around is super cumbersome. I'm imagining something like zeroconf for wireless devices.

I tried using the azure portal UI and disliked it intensely. It has no bulk editing of app permissions and it's really hard to see at a glance what parameters are set on a given app so I can clone it in other  test environments. So, I am trying to build my own automation scripts so I never have to look at it again. The process is quite cumbersome. In the old days, I would open up excel, access, etc and do things and record macros and then customize them. I'm amazed azure doesn't behave like this. It's like all the best features of the old-school microsoft have been foresaken. Imagine if one could use the azure portal UI and record powershell macros. 

Another disappointment is, being a [C# shop](https://www.cognivasolutions.com/) and MSFT partner that are migrating our VSTO addins to office 365, we are now being forced to use powershell because not all functions are being supported outside powershell, nor are there the SDKs of old so we can code against azure with plain functions in managed code. Powershell may be an upgrade for system admins but it's not type-safe, has an inferior debug experience with poor intellisense and no resharper, and you can't unload appdomains when talking to managed code during debugging in the ISE. This whole hodgepodge of technologies is really starting to grate on my nerves, and making suggestions like this via uservoice means it goes into an issue tracker black hole. So I figured i'd rant here and perhaps get some eyes on my frustration...

**Comment by jmprieur** [comment]
@xstos:  Did you try the Microsoft Graph API? 
https://docs.microsoft.com/en-us/graph/api/resources/application?view=graph-rest-1.0

**Comment by xstos** [comment]
Thanks, I had tried it in the past, but couldn't quite the authentication working and I was getting errors like "Insufficient privileges to complete the operation." This powershell sample was just so much easier to get going. 

All it took was

$username = "username"
$password = convertTo-securestring 'Password' -AsPlainText -Force; 
$cred = new-object -typename System.Management.Automation.PSCredential -argumentlist $username, $password; 
Connect-AzureRmAccount -Credential $cred;

However, I just setup another test up in VS:

`string[] scopes = {"user.read"};
var app = ConfidentialClientApplicationBuilder
	.Create(graphClientId)
	.WithTenantId(tenantId)
	.WithClientSecret("...")
	.Build();

ClientCredentialProvider authProvider = new ClientCredentialProvider(app);
var graphClient = new GraphServiceClient(authProvider);
SecureString ss = new SecureString();
foreach (var c in Password) ss.AppendChar(c);
var application = await graphClient.Applications
	.Request()
	.WithUsernamePassword(Username, ss)
	.GetAsync();`

and it seems to be working. I guess I gave up on it too quickly when I last tried this a year or two ago. 

Thanks for the help. I didn't realize the graph api could do app registrations... I was under the impression it was way more limited than what I'm seeing there.
